csp_report.php

```diff
@@ -76,26 +84,65 @@
 $table->sortable(true, 'failcounter', SORT_DESC);
 $table->define_columns(array(
     'failcounter',
-    'documenturi',
-    'blockeduri',
     'violateddirective',
-    'timecreated',
+    'blockeduri',
+    'highestviolaters',
     'timeupdated',
     'action',
 ));
 $table->define_headers(array(
     $failcounter,
-    $documenturi,
-    $blockeduri,
     $violateddirective,
-    $timecreated,
+    $blockeduri,
+    $highestviolaters,
     $timeupdated,
     $action,
 ));
 
-$fields = 'id, sha1hash, documenturi, blockeduri, violateddirective, failcounter, timecreated, timeupdated';
-$from = '{local_csp}';
-$where = '1 = 1';
+// If user has clicked on a violation to view all violation entries
+if (($viewviolationclass = optional_param('viewviolationclass', false, PARAM_TEXT)) !== false && confirm_sesskey()) {
```

Comment:
I'm not keen on the long combined & sesskey, as it won't actually fail is the sesskey is wrong and it will keep doing something else. I think its easier to read and follow if you say:

```
$viewviolationclass = optional_param('viewviolationclass', false, PARAM_TEXT);

if ($viewviolationclass !== false) {
    require_sesskey();
```